### PR TITLE
Added the update of resources for thirdparty

### PR DESCRIPTION
**Pull request by pragamj**
Signed-off-by: pragam <pragam.jain@seagate.com>

# Problem Statement
-Added the update of resources for thirdparty

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
[deployment.log](https://github.com/Seagate/cortx-test/files/8685811/deployment.log)
[TEST-N-NODE_TestContDeployment_20220513031335.log](https://github.com/Seagate/cortx-test/files/8685812/TEST-N-NODE_TestContDeployment_20220513031335.log)

-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)
[collection_log.txt](https://github.com/Seagate/cortx-test/files/8685810/collection_log.txt)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 1]
```diff
@@ -1790,3 +1794,43 @@ def namespace_name_generator(size):
...
+            resource = common['resource_allocation']
+            consul = resource['consul']
+            zookeeper = resource['zookeeper']['resources']
+            kafka = resource['kafka']['resources']
+            type_list = ['requests', 'limits']
+            consul_list = ['server', 'client']
+            third_party_resource = self.deploy_cfg['thirdparty_resource']
+            # updating the consul server /client request and limit resources
+            for elem in consul_list:
+                for res_type in type_list:
```
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:1818]: can we have type_list as outer for loop
Consul server,clients can be traversed in inner loop

File: config/prov/prov_test.yaml [hunk 2]
```diff
@@ -186,6 +186,35 @@ k8s_cortx_deploy:
     deploy: 7200
     destroy: 900
     status: 120
+  thirdparty_resource:
+    server:
```
Comment (priyankaborawake) [inline config/prov/prov_test.yaml:190]: Can you add the consul here(can be done as comment)

File: libs/prov/prov_k8s_cortx_deploy.py (hunk 1 above)
Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py]: done

File: config/prov/prov_test.yaml (hunk 2 above)
Comment (pragamj) [reply config/prov/prov_test.yaml]: done

**Review by priyankaborawake — APPROVED** [review]

**Review by ketanarlulkar — APPROVED** [review]

**Review by seagate-sarang-sawant — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Comment by sonalk0209** [comment]
Merging as all approvals are met.

**Pull request merged**